### DISCUSSION TITLE
How to Hide Syncfusion License Key

### DISCUSSION THREAD

**Discussion by jgyo** [community/community#145684] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have a project that uses Syncfusion libraries. To compile runnable code from the source, I must have a license key in one of the source files. I want to exclude that key from the GitHub repository, but I want a stub file to replace it where I can include a "SYNCFUSION LICENSE KEY GOES HERE" in place of the key so that users can borrow the code and know how to incorporate their key.

I am struggling with that, though. If I include a line in .gitignore for the file, sometimes it is not excluded, and other times, it deletes the destination file with the same name.

How can I manage this issue?

**Comment by kivanc57** [comment, marked as answer]
Then you need to create an `.env` file where you usually store all the secrets and keys if you work on local machine. After that you can access these secrets as constants in your script. Although, the way you access may alter based on the language you are using but it's usually very simple.

If you whether automate it or work somerhow on remote there is also a link how to add the secrets to your repository:
https://docs.github.com/en/codespaces/managing-codespaces-for-your-organization/managing-development-environment-secrets-for-your-repository-or-organization

**Reply by jgyo** [reply]
I've looked at that page before, but AFAIK, it will not work with the way Syncfusion licenses are registered. They require the license key to be embedded in the code so that it can be used as a parameter to the register method when the program runs on the user's computer. So, it needs to be in my code when I compile but not when I commit the code to the server.
